### PR TITLE
Add jwt authentication.

### PR DESCRIPTION
**Pull request by DevMDamien**
I don't know what I did when I rebased but seems working.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,7 @@
 .phpunit
 /phpunit.xml
 ###< symfony/phpunit-bridge ###
+
+###> lexik/jwt-authentication-bundle ###
+#%CONFIG_DIR%/jwt/*.pem
```
Comment (ro0NL) [inline .gitignore]: what about adding `bin/generate-jwt` instead? and keep ignoring the pem files, which is what the bundle does by default AFAIK: https://github.com/symfony/recipes/blob/master/lexik/jwt-authentication-bundle/2.3/manifest.json#L16

File: config/routes.yaml [hunk 2]
```diff
@@ -1,3 +1,6 @@
 index:
     path: /
     controller: App\Controller\DefaultController
+
+api_login_check:
+    path: /api/login_check
```
Comment (ro0NL) [inline config/routes.yaml]: i'd say `routes/api.php` for now :)

File: config/routes/msgphp_user.php [hunk 3]
```diff
@@ -21,5 +21,7 @@
             ->controller(Controller\User\MyAccountController::class)
         ->add('email_confirm', '/my-account/confirm-email/{token}')
             ->controller(Controller\User\ConfirmEmailController::class)
+        ->add('api_index', '/api/test-connection')
+            ->controller(Controller\Api\DefaultController::class)
```
Comment (ro0NL) [inline config/routes/msgphp_user.php]: also `routes/api.php`

File: .gitignore [hunk 4]
```diff
@@ -11,3 +11,7 @@
 .phpunit
 /phpunit.xml
 ###< symfony/phpunit-bridge ###
+
+###> lexik/jwt-authentication-bundle ###
+%CONFIG_DIR%/jwt/*.pem
```
Comment (ro0NL) [inline .gitignore]: shouldnt this be `/config/jwt/*.pem` :thinking:

File: bin/generate-jwt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+$pubKeyData = openssl_pkey_get_details($pKeyRes);
+
+$fs = new Filesystem();
+$fs->mkdir(dirname(__DIR__.'/../'.$_SERVER['JWT_PUBLIC_KEY_PATH']));
+$fs->mkdir(dirname(__DIR__.'/../'.$_SERVER['JWT_PRIVATE_KEY_PATH']));
+
+$fs->dumpFile(__DIR__.'/../'.$_SERVER['JWT_PUBLIC_KEY_PATH'], $pubKeyData['key']);
+$fs->dumpFile(__DIR__.'/../'.$_SERVER['JWT_PRIVATE_KEY_PATH'], $privKeyData);
+
+echo "Keys successfully generated";
```
Comment (ro0NL) [inline bin/generate-jwt]: could you add trailing "\n" (`... generated\n";`)

+ newline on end of file :)

File: config/routes/api.php [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+use App\Controller;
+
+use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+
+return function (RoutingConfigurator $routes) {
+    $routes
+        ->add('api_index', '/api/test-connection')
+            ->controller(Controller\Api\DefaultController::class)
+        ->add('api_login_check', '/api/login_check')
```
Comment (ro0NL) [inline config/routes/api.php]: `/api/login-check` pls :)

File: .gitignore (hunk 4 above)
Comment (DevMDamien) [reply .gitignore]: It's from the recipe.

Comment (ro0NL) [reply .gitignore]: on it :D

Comment (ro0NL) [reply .gitignore]: https://github.com/symfony/recipes/pull/386

Comment (ro0NL) [reply .gitignore]: @DevMDamien can you re-install the recipe once more. I believe Flex handles the placeholder now.

**Comment by ro0NL** [comment]
Thx @DevMDamien added to https://github.com/msgphp/symfony-demo-app/pull/28 :+1:

**Pull request merged**